### PR TITLE
feat: expose current_patch_number from updater

### PR DESCRIPTION
**Pull request by bryanoltman**

### CODE REVIEW

File: shell/platform/android/android_exports.lst [hunk 1]
```diff
@@ -11,6 +11,7 @@
     _binary_icudtl_dat_size;
     
     shorebird_check_for_update;
+    shorebird_current_boot_patch_number;
```
Comment (felangel) [inline shell/platform/android/android_exports.lst:14]: I thought next_boot_patch_number was the one the device would boot from? How would that differ from the new api?

Comment (bryanoltman) [reply shell/platform/android/android_exports.lst:14]: This tells what patch the app is _currently_ running on. If a new patch is downloaded, `shorebird_next_boot_patch_number` will be the number of that patch, not the one that is currently running.

Comment (eseidel) [reply shell/platform/android/android_exports.lst:14]: next_boot_patch_number is "on next boot I will try patch X"
current_boot_patch_number is "I'm currently booted from X"

I presume this is for displaying the current full version info?

Comment (felangel) [reply shell/platform/android/android_exports.lst:14]: Okay makes sense. So initially both current and next would return 0? When a new patch is downloaded, current would still be 0 and next would be 1? When the new patch has booted current and next would both be 1?

Comment (bryanoltman) [reply shell/platform/android/android_exports.lst:14]: > I presume this is for displaying the current full version info?

Yep! The idea is "You're on version 1.0.0+1 (3), version 1.0.0+1 (5) is available" or similar

Comment (bryanoltman) [reply shell/platform/android/android_exports.lst:14]: > Okay makes sense. So initially both current and next would return 0? When a new patch is downloaded, current would still be 0 and next would be 1? When the new patch has booted current and next would both be 1?

That's correct – if no new patch is available, these numbers will be the same.

**Review by eseidel — APPROVED** [review]

**Pull request merged**